### DISCUSSION TITLE
hyperlinks to the bibliography fails

### DISCUSSION THREAD

**Discussion by wisejky** [quarto-dev/quarto-cli#9238] [Q&A] [answered]
### Description

I encountered an issue with citations while writing my paper using Quarto. After citing references, the generated PDF did not include hyperlinks to the bibliography; instead, the references appeared as regular text.
I will present a comparison before and after:

````
[@amihud2002b]  
@amihud2002b
````

![image](https://github.com/quarto-dev/quarto-cli/assets/120883966/1623aad9-066a-4b90-9f98-11fa8f339d9d)

I searched the community for related issues, such as the setting for `link-citations: true`.（Although everyone advises against it.）

![image](https://github.com/quarto-dev/quarto-cli/assets/120883966/2a2d1bba-e0c8-40a0-baad-d3374bfc8d6f)

Then I went to look for the process files.
in tex：
````
% definitions for citeproc citations 
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}

...

(\citeproc{ref-amihud2002b}{Amihud 2002}) 
Amihud (\citeproc{ref-amihud2002b}{2002})

...

\phantomsection\label{refs}
\begin{CSLReferences}{1}{0}

\bibitem[\citeproctext]{ref-amihud2002b}
Amihud, Yakov. 2002. {``Illiquidity and Stock Returns: Cross-Section and
Time-Series Effects.''} \emph{Journal of Financial Markets} 5 (1):
31--56. \url{https://doi.org/10.1016/S1386-4181(01)00024-6}.

\end{CSLReferences}
````

I use [nmfs-opensci/quarto-thesis](https://github.com/nmfs-opensci/quarto-thesis) as my setup.

TKS you all!!!

below is my ideal output: 
![image](https://github.com/quarto-dev/quarto-cli/assets/120883966/b090711f-5808-4cfb-82f4-5ff204a6a44e)

**Comment by mcanouil** [comment]
Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

Additionally, since it seems to be a LaTeX issue, please ensure the issue is not with the format extension you are using, i.e., try using default from Quarto.
Also, to debug LaTeX, you can use `keep-tex` and look at the intermediate LaTeX document.



---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by wisejky** [reply]
thanks for your reply!

by your hints, i step by step check my related latex code, and finally find that:

if a drop these(which is on the head of my file):

````latex
\usepackage[utf8]{inputenc} % Required for inputting international characters
\usepackage[backend=bibtex, style=authoryear, natbib=true]{biblatex} % Use the bibtex backend with the authoryear citation style (which resembles APA)
\usepackage[autostyle=true]{csquotes} % Required to generate language-dependent quotes in the bibliography
````

then everyting goes well...... I don't know why but hyperlinks to the bibliography works!

![image](https://github.com/quarto-dev/quarto-cli/assets/120883966/de72e812-5413-48e8-b51b-3276e17407bb)

**Reply by mcanouil** [reply, marked as answer]
That's the "magic" thing with LaTeX unfortunately.
There might be a conflict with options.
Unfortunately, when using non default, Quarto cannot guarantee anything and it becomes up to you to figure out the issue and to solve it.
Additionally, you can look at the partials/template used by Quarto to help you figure out the solution which might be to replace the partials, see <https://quarto.org/docs/journals/templates.html#latex-partials>.

**Reply by wisejky** [reply]
thanks for your useful suggestion，i would keep learning on it

**Comment by cderv** [comment]
@wisejky it seems this could be an issue with the template you are using.

If they really add  
````
\usepackage[backend=bibtex, style=authoryear, natbib=true]{biblatex}
````

in the template unconditionally, I do think this means that `biblatex` should be used as processing engine 
So `cite-method` should be changed (https://quarto.org/docs/output-formats/pdf-basics.html#citations) - the default will be Pandoc citeproc. 

`link-citation: true` is required only for Pandoc citeproc. If you set `cite-method: natbib` or `cite-method: biblatex`, I believe there will be link created anyway. 

And by using those citation method, especially the latter, I don't there will be conflict with the options set directly in LaTeX 

Hope it helps

**Reply by wisejky** [reply]
YES! 

````
\usepackage[backend=bibtex, style=authoryear, natbib=true]{biblatex}
````
 
should be droped!

Thanks a lot
